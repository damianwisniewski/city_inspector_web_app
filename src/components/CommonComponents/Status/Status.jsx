import React from 'react'
import PropTypes from 'prop-types'

// SCSS
import './Status.scss'
import iconAssets from '../../../assets/styleModules/icons.module.scss'

const Status = ({ id, message, type, position }) => (
	<p
		id={id}
		className={`status-component status-component--${type} status-component--${position} ${
			iconAssets[type]
		}`}
	>
		{message}
	</p>
)

Status.defaultProps = {
	message: '',
	position: 'horizontal',
}

Status.propTypes = {
	message: PropTypes.string,
	type: PropTypes.oneOf(['error', 'warning', 'correct']).isRequired,
	position: PropTypes.oneOf(['horizontal', 'vertical']),
}

export default Status
