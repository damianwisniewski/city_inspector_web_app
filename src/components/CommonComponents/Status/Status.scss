@import '../../../assets/styles/index';

.status-component {
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: inherit;
	color: inherit;

	&::before {
		font-size: inherit;
	}

	&--error::before {
		color: $red;
	}

	&--warning::before {
		color: $yellow;
	}

	&--correct::before {
		color: $green;
	}

	&--horizonal {
		flex-direction: row;
	}

	&--vertical {
		flex-direction: column;
	}
}
