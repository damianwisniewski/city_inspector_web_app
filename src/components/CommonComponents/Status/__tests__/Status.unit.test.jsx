import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Status from '../Status'

configure({ adapter: new Adapter() })

describe('<Status />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let StatusElement
	const StatusText = 'ExampleText'
	const id = 'mockId'

	test('should render correctly Status type error', () => {
		StatusElement = shallow(<Status id={id} type='error' message={StatusText} />)
		expect(toJson(StatusElement)).toMatchSnapshot()
	})

	test('should render correctly Status type warning', () => {
		StatusElement = shallow(<Status id={id} type='warning' message={StatusText} />)
		expect(toJson(StatusElement)).toMatchSnapshot()
	})

	test('should render correctly Status type correct', () => {
		StatusElement = shallow(<Status id={id} type='correct' message={StatusText} />)
		expect(toJson(StatusElement)).toMatchSnapshot()
	})

	test('should render correctly Status with position horizontal', () => {
		StatusElement = shallow(
			<Status id={id} type='correct' position='horizontal' message={StatusText} />,
		)
		expect(toJson(StatusElement)).toMatchSnapshot()
	})

	test('should render correctly Status with position vertical', () => {
		StatusElement = shallow(
			<Status id={id} type='correct' position='vertical' message={StatusText} />,
		)
		expect(toJson(StatusElement)).toMatchSnapshot()
	})
})
