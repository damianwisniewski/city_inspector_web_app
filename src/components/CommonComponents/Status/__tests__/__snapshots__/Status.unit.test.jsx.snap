// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render correctly Status type correct 1`] = `
<p
  className="status-component status-component--correct status-component--horizontal correct"
  id="mockId"
>
  ExampleText
</p>
`;

exports[`<Status /> should render correctly Status type error 1`] = `
<p
  className="status-component status-component--error status-component--horizontal error"
  id="mockId"
>
  ExampleText
</p>
`;

exports[`<Status /> should render correctly Status type warning 1`] = `
<p
  className="status-component status-component--warning status-component--horizontal warning"
  id="mockId"
>
  ExampleText
</p>
`;

exports[`<Status /> should render correctly Status with position horizontal 1`] = `
<p
  className="status-component status-component--correct status-component--horizontal correct"
  id="mockId"
>
  ExampleText
</p>
`;

exports[`<Status /> should render correctly Status with position vertical 1`] = `
<p
  className="status-component status-component--correct status-component--vertical correct"
  id="mockId"
>
  ExampleText
</p>
`;
